### PR TITLE
Migrate "beam_PostCommit_Python_Nexmark_Direct" Jenkins jobs to GitHub Actions

### PR DESCRIPTION
**Pull request by andreydevyatkin**
The following jenkins job has been migrated:
- https://ci-beam.apache.org/job/beam_PostCommit_Python_Nexmark_Direct/

### How has it been tested?

The jobs have been tested on Self-hosted GKE runners ubuntu 20.04.
The following test cases are passed:

- by manually running the job (`workflow_dispatch`)

Screenshot 2023-09-15 at 20 22 11

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

File: .github/workflows/beam_PostCommit_Python_Nexmark_Direct.yml [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+jobs:
+  beam_PostCommit_Python_Nexmark_Direct:
+    name: ${{matrix.job_name}} (${{matrix.job_phrase}})
+    runs-on: [self-hosted, ubuntu-20.04, main]
+    timeout-minutes: 240
+    strategy:
+      fail-fast: false
+      matrix:
+        job_name: [beam_PostCommit_Java_Nexmark_Direct]
+        job_phrase: [Run Python Direct Runner Nexmark Tests]
```
Comment (damccorm) [inline .github/workflows/beam_PostCommit_Python_Nexmark_Direct.yml:99]: I'm confused by the mixing of Java/Python in this job; why is one of these Java and one is Python?

Similarly, it seems like we're running both Java and Python benchmarks? Should those be in different workflow files?

Comment (andreydevyatkin) [reply .github/workflows/beam_PostCommit_Python_Nexmark_Direct.yml:99]: I didn't dig too deep into the intent, but based on the implementation and Jenkins runs I see that both tasks are being executed - https://ci-beam.apache.org/job/beam_PostCommit_Python_Nexmark_Direct/1003/consoleFull

File: .github/workflows/beam_PostCommit_Python_Nexmark_Direct.yml [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        query: [0, 2, 3, 5, 7, 8, 10, 11]
```
Comment (damccorm) [inline .github/workflows/beam_PostCommit_Python_Nexmark_Direct.yml:100]: Could we add the links we have here for context on why we're only running some of the queries - https://github.com/apache/beam/blob/1f980eaa894cc43ea5ca1aeb4cb2ef1de1162b17/.test-infra/jenkins/NexmarkBuilder.groovy#L187

File: .github/workflows/beam_PostCommit_Python_Nexmark_Direct.yml (hunk 1 above)
Comment (damccorm) [reply .github/workflows/beam_PostCommit_Python_Nexmark_Direct.yml:99]: Ok, the job name should still be Python, though, right?

Comment (andreydevyatkin) [reply .github/workflows/beam_PostCommit_Python_Nexmark_Direct.yml:99]: Agree, it's confusing, but I see that Jenkins uses this exact comment phrase https://github.com/apache/beam/blob/master/.test-infra/jenkins/job_PostCommit_Python_Nexmark_Direct.groovy#L38.

File: .github/workflows/beam_PostCommit_Python_Nexmark_Direct.yml (hunk 2 above)
Comment (andreydevyatkin) [reply .github/workflows/beam_PostCommit_Python_Nexmark_Direct.yml:102]: Done, thanks!

File: .github/workflows/beam_PostCommit_Python_Nexmark_Direct.yml (hunk 1 above)
Comment (damccorm) [reply .github/workflows/beam_PostCommit_Python_Nexmark_Direct.yml:99]: I'm talking about the job_name, not the job_phrase (`beam_PostCommit_Java_Nexmark_Direct` should be `beam_PostCommit_Python_Nexmark_Direct` I think)

File: .github/workflows/beam_PostCommit_Python_Nexmark_Direct.yml (hunk 2 above)
Comment (damccorm) [reply .github/workflows/beam_PostCommit_Python_Nexmark_Direct.yml:102]: Sorry, I meant the github issue links that are referenced in that file. I think the Jenkins file should go away eventually, so we should duplicate the github issue links here.

Comment (andreydevyatkin) [reply .github/workflows/beam_PostCommit_Python_Nexmark_Direct.yml:104]: Updated

**Review by damccorm — APPROVED** [review]
Thanks!

**Pull request merged**